### PR TITLE
filen-desktop: init at 2.0.22

### LINKED ISSUE #213628: add Filen Desktop
**Project description**
Sync files and folders to all devices connected to your account. With support for more than 5 different sync modes your files will always be up to date.

**Metadata**

* homepage URL: https://github.com/FilenCloudDienste/filen-desktop
* source URL: https://github.com/FilenCloudDienste/filen-desktop
* license: AGPL-3.0
* platforms: Linux

### PR DESCRIPTION
**Pull request by taj-ny**
###### Description of changes
https://github.com/FilenCloudDienste/filen-desktop

I wasn't able to build the application from source.

closes #213628

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/filen-desktop/default.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{ appimageTools
+, fetchurl
+, lib
+, stdenv
+, undmg
+}:
```
Comment (AndersonTorres) [inline pkgs/applications/misc/filen-desktop/default.nix:6]:
```suggestion
{ lib
, stdenv
, fetchurl
, appimageTools
, undmg
}:
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Comment by AndersonTorres** [comment]
?

**Comment by tien-vo** [comment]
Looks like this was dropped? Is anyone still working on it?

**Comment by AndersonTorres** [comment]
Yes. No.

**Comment by JManch** [comment]
Here's a minimal working `x86-64-linux` package for anyone who really needs file-desktop. The main addition to this PR is adding the `libappindicator-gtk3` library. Without it, the tray doesn't work and filen is unusable.

``` nix
{ lib
, fetchurl
, makeWrapper
, appimageTools
, libappindicator-gtk3
}:
let
  pname = "filen-desktop";
  version = "2.0.24";

  src = fetchurl {
    # Package will need a hash update whenever upstream updates
    # https://github.com/FilenCloudDienste/filen-desktop/issues/208
    url = "https://cdn.filen.io/desktop/release/filen_x86_64.AppImage";
    hash = "sha256-5vkndT9V/81fUdzS+KTfAjPAGO0IJRx8QhNxBNG8nnU=";
  };

  appimageContents = appimageTools.extractType2 { inherit pname version src; };

  # Needed for the tray to function
  libPath = lib.makeLibraryPath [ libappindicator-gtk3 ];
in
appimageTools.wrapType2 rec {
  inherit pname version src;

  extraInstallCommands = ''
    mv $out/bin/${pname}-${version} $out/bin/${pname}

    source "${makeWrapper}/nix-support/setup-hook"
    wrapProgram $out/bin/filen-desktop \
      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform=wayland --enable-features=WaylandWindowDecorations}}" \
      --prefix LD_LIBRARY_PATH : ${libPath}

    install -m 444 -D ${appimageContents}/${pname}.desktop $out/share/applications/${pname}.desktop
    substituteInPlace $out/share/applications/${pname}.desktop \
      --replace 'Exec=AppRun' 'Exec=filen-desktop'
    cp -r ${appimageContents}/usr/share/icons $out/share
  '';

  meta = with lib; {
    description = "Desktop client for Filen.io";
    homepage = "https://filen.io/";
    license = licenses.agpl3Plus;
    changelog = "https://github.com/FilenCloudDienste/filen-desktop/releases/tag/v${version}";
    platforms = [ "x86_64-linux" ];
  };
}
```

I'm not going to PR this because the URL format filen uses makes it impossible to statically reference the latest version https://github.com/FilenCloudDienste/filen-desktop/issues/208. This means the package will break whenever filen updates and, I assume, makes it unsuitable for nixpkgs. Personally, I'm not interested in always using the version prior to the latest.

**Comment by npnpatidar** [comment]
`{ lib
, fetchurl
, makeWrapper
, appimageTools
, libappindicator-gtk3
}:
let
  pname = "filen-desktop";
  version = "latest";

  src = pkgs.fetchurl {
    url =
      "https://cdn.filen.io/@filen/desktop/release/latest/Filen_linux_x86_64.AppImage";
    sha256 = "sha256-Nao5By8Z8lMbRcp2Mgw+xaiiFzUxCm6S3SAE5FfDZpk=";
  };

  appimageContents = appimageTools.extractType2 { inherit pname version src; };
 name = "filendesktop";
  # Needed for the tray to function
  libPath = lib.makeLibraryPath [ libappindicator-gtk3 ];
in
appimageTools.wrapType2 rec {
  inherit pname version src name;

  extraInstallCommands = ''
     source "${pkgs.makeWrapper}/nix-support/setup-hook"
    wrapProgram $out/bin/${pname} \
      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform=wayland --enable-features=WaylandWindowDecorations}}" \
      --prefix LD_LIBRARY_PATH : ${libPath}

    install -m 444 -D ${appimageContents}/@${name}.desktop $out/share/applications/${pname}.desktop
    substituteInPlace $out/share/applications/${pname}.desktop \
      --replace 'Exec=AppRun' 'Exec=filen-desktop'
    cp -r ${appimageContents}/usr/share/icons $out/share
  '';

  meta = with lib; {
    description = "Desktop client for Filen.io";
    homepage = "https://filen.io/";
    license = licenses.agpl3Plus;
    changelog = "https://github.com/FilenCloudDienste/filen-desktop/releases/tag/v${version}";
    platforms = [ "x86_64-linux" ];
  };
}`


Now we can statistically reference latest version. They  have changed the url

**Pull request closed without merging**